### PR TITLE
Subscriptions, mutations, and the `counter` example

### PR DESCRIPTION
**Pull request by Twey**
Atop https://github.com/linera-io/linera-web/pull/10.

Adds support for subscriptions and mutations, as well as a frontend for the `linera-protocol` `counter` example that demonstrates the usage of both.

### CODE REVIEW

File: client/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+    client_options::ClientOptions,
+    wallet::Wallet,
+};
+use linera_core::node::{ValidatorNode as _, ValidatorNodeProvider as _};
+use linera_views::store::WithError;
+use serde::ser::Serialize as _;
+use wasm_bindgen::prelude::*;
+use web_sys::{js_sys, wasm_bindgen};
+
+// TODO(12): convert to IndexedDbStore once we refactor Context
```
Comment (ma2bd) [inline client/src/lib.rs:23]: nit: We've used `TODO(#12)` so far in the other repo.

File: client/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+    /// If storage is inaccessible.
+    #[wasm_bindgen]
+    pub async fn read() -> Result<Option<JsWallet>, JsError> {
+        Ok(None)
+    }
+}
+
+/// The full client API, exposed to the wallet implementation.  Calls
+/// to this API can be trusted to have originated from the user's
+/// request.  This struct is the backend for the extension itself
```
Comment (afck) [inline client/src/lib.rs:98]: Nit:

```suggestion
/// request. This struct is the backend for the extension itself
```

(Also elsewhere.)

File: client/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "scripts": {
+        "install": "true",
+        "build": "bash build.bash",
+        "prepare": "pnpm build",
+        "lint": "cargo fmt --check && cargo clippy -- -W clippy::pedantic",
+        "ci": "pnpm lint"
+    },
+    "keywords": [],
+    "author": "",
+    "license": "ISC"
```
Comment (jvff) [inline client/package.json:16]: Should we populate this, or are we planning on doing that later?

File: client/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+            }
+        });
+    }
+
+    async fn default_chain_client(&self) -> Result<ChainClient, JsError> {
+        let client_context = self.client_context.lock().await;
+        let chain_id = client_context
+            .wallet()
+            .default_chain()
+            .expect("No default chain");
```
Comment (jvff) [inline client/src/lib.rs:175]: Optional nit:

```suggestion
            .expect("A default chain should be configured");
```

**Review by jvff — APPROVED** [review]
This is awesome! :tada:

File: examples/counter/.gitignore [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Logs
+logs
```
[the remark this answers is not in the document]
Comment (Twey) [reply examples/counter/.gitignore]: Actually we already have all these rules there!  This is just the `.gitignore` generated by Vite.  I've removed it.

File: client/package.json (hunk 3 above)
Comment (Twey) [reply client/package.json]: No time like the present!

**Pull request merged**